src/sentry/api/bases/external_actor.py

```diff
@@ -40,11 +41,26 @@
 }
 
 
+class ExternalActorResponse(TypedDict):
+    id: int
+    external_id: str | None
+    external_name: str
+    provider: int
+    integration_id: int
+
+
 class ExternalActorSerializerBase(CamelSnakeModelSerializer):
-    external_id = serializers.CharField(required=False, allow_null=True)
-    external_name = serializers.CharField(required=True)
-    provider = serializers.ChoiceField(choices=get_provider_choices(AVAILABLE_PROVIDERS))
-    integration_id = serializers.IntegerField(required=True)
+    external_id = serializers.CharField(
+        required=False, allow_null=True, help_text="The associated User ID for provider."
+    )
+    external_name = serializers.CharField(
+        required=True, help_text="The associated username for the provider."
+    )
+    provider = serializers.ChoiceField(
+        choices=get_provider_choices(AVAILABLE_PROVIDERS),
+        help_text="The provider. ('github', 'gitlab', 'slack')",
+    )
```

Comment:
This is a bit hard to document b/c we have the setting that automatically generates bullet points from choices, so it'll include more than github, gitlab, and slack which you can see in your screenshot

Not sure how to avoid this tbh

Reply:
actually ill remove the list i sent altogether.  even though we don't really officially use the ExternalActor table for other providers than the 3 i listed, technically we can add this linking for all of it. this will eventually help ecosystem anyway as we start supporting team linking in the other messaging and scm integrations.

Reply:
so the docs will include all providers, even tho we only support some of them?

Reply:
we support all of them, but we just don't do anything with them.

Reply:
Hmm should we change the `get_provider_choices` to be limited to those 3 providers then?

Reply:
good point, let me do that

Reply:
also can the choicefield just accept a regular list of strings, instead of a list of tuples? That way what's generated won't include the second item in the tuple which doesn't provide any useful info

Reply:
actually i don't want to limit it to those 3. in code technically we support other integrations as [well](https://github.com/getsentry/sentry/blob/27195bd74d86d77bbb37e7f962aabfd2a3c398b7/src/sentry/integrations/msteams/notifications.py#L88). the comment that existed on the old endpoint predated the new changes, it should have been deleted.

Reply:
actually I don't know why it's autogenerating those bullet points.

It's controlled by the `ENUM_GENERATE_CHOICE_DESCRIPTION` setting in `SPECTACULAR_SETTINGS` in `server.py`, but it's not there anymore

Reply:
oh actually I think it just defaults to True always. If the choicefield just has a regular list, it copies the first value as both the choice value - choice name